src/impl/peerconnection.cpp

```diff
@@ -88,6 +83,11 @@ void PeerConnection::close() {
 	mProcessor.enqueue(&PeerConnection::closeTracks, shared_from_this());
 
 	closeTransports();
+
+	// We must join all tasks here, in order to ensure all references to 'this' are freed.
+	// otherwise the destructor will never be called, causing async callbacks to fire after the base
+	// PeerConnection was freed.
+	mProcessor.join();
```

Comment:
Joining here will make the processor wait for itself and therefore create a deadlock when the user calls `close()` from a callback (for instance on `State::Failed`).